src/flask/json/__init__.py

```diff
@@ -45,6 +46,8 @@ def default(self, o: t.Any) -> t.Any:
         overriding how basic types like ``str`` or ``list`` are
         serialized, they are handled before this method.
         """
+        if isinstance(o, decimal.Decimal):
+            return str(o)
```

Comment:
I'm not sure that's correct.. simplejson converts to a number, while your code converts to a string.

Reply:
okay sure, I didnt think it through, I'll make it an `int`.